games/minimal/mods/test/player.lua

```diff
@@ -19,9 +31,18 @@ local function run_hpchangereason_tests(player)
 	expect = { df = 3458973454, type = "fall", from = "mod" }
 	player:set_hp(10, { type = "fall", df = 3458973454 })
 	assert(expect == nil)
+
+	tests_enabled = false
+
+	--
+	-- Tests end
+	--
+
+	-- Restore old HP
+	player:set_hp(old_hp)
 end
 minetest.register_on_player_hpchange(function(player, hp, reason)
-	if not expect then
+	if not tests_enabled then
```

Comment:
Note: Checking for `not expect` is now redundant, because `not tests_enabled` will evaluate to `false` once the tests are over (L35), and the rest of the callback would be ignored henceforth. Btw, I've tested this by adding a `chat_send_all` after this `if` block.